parse/analyze.go

```diff
@@ -747,12 +755,31 @@ func (s *sqlAnalyzer) VisitExpressionArrayAccess(p0 *ExpressionArrayAccess) any
 		s.errs.AddErr(p0, ErrAssignment, "array access is not supported in in-line action statements")
 	}
 
-	idxAttr, ok := p0.Index.Accept(s).(*types.DataType)
-	if !ok {
-		return s.expressionTypeErr(p0.Index)
-	}
-	if !idxAttr.Equals(types.IntType) {
-		return s.typeErr(p0.Index, idxAttr, types.IntType)
+	var isArray bool
+	if p0.Index != nil {
+		// if single index, result is not an array
+		idxAttr, ok := p0.Index.Accept(s).(*types.DataType)
+		if !ok {
+			return s.expressionTypeErr(p0.Index)
+		}
+		if !idxAttr.Equals(types.IntType) {
+			return s.typeErr(p0.Index, idxAttr, types.IntType)
+		}
+	} else {
+		// if multiple indexes, result is an array
+		isArray = true
+		for _, idx := range p0.FromTo {
```

Comment:
I think we need one more check, `leftIndex < rightIndex`, but it seems we cannot enforce it here?

Reply:
1. We can't enforce that
2. Postgres actually allows this, so it is a non-issue

Reply:
cool, then it's fine